kernel-spark/src/main/java/io/delta/kernel/spark/snapshot/ManagedCommitClient.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (2025) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.spark.snapshot;
+
+import io.delta.kernel.CommitRange;
+import io.delta.kernel.Snapshot;
+import io.delta.kernel.engine.Engine;
+import io.delta.kernel.internal.files.ParsedLogData;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Catalog-managed commit client that knows how to load snapshots and commit ranges for a specific
+ * table.
+ */
+public interface ManagedCommitClient extends AutoCloseable {
```

Comment:
Maybe not called it Client？This abstraction is 1 level above to encapsulate specific Catalog implementation, maybe call it `ManagedCatalogAdapter`?